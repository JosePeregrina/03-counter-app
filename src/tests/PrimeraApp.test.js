import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('El componente PrimeraApp', () => {
  /* test('debe de mostrar el mensaje "Hola, Soy Goku"', () => {
    const saludo = 'Hola, Soy Goku';
     La manera sin desestructuraci√≥n
     const wrapper = render(<PrimeraApp saludo={saludo}/>)
     wrapper.getByText()

    const { getByText } = render(<PrimeraApp saludo={saludo} />);
    expect(getByText(saludo)).toBeInTheDocument();
  }); */

  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola, Soy Goku';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el subtitulo que se mande por props', () => {
    const saludo = 'Hola soy goku';
    const subtitulo = 'Soy un guerrero Z'
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

    const textoParrafo = wrapper.find('p').text();

    expect( textoParrafo ).toBe(subtitulo);

  });
});
