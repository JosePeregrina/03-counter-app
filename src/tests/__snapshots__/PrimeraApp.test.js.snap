// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente PrimeraApp debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
